### DISCUSSION TITLE
How to use RetrieveUserProxyAgent with Pinecone database?

### DISCUSSION THREAD

**Discussion by SwastikGorai** [microsoft/autogen#908] [Q&A]
Can i use pinecone database as it is done here for the case of qdrant ? 
```
ragproxyagent = QdrantRetrieveUserProxyAgent(
    name="ragproxyagent",
    human_input_mode="NEVER",
    max_consecutive_auto_reply=10,
    retrieve_config={
        "task": "code",
        "docs_path": "~/path/to/FLAML/website/docs/reference", 
        "chunk_token_size": 2000,
        "model": config_list[0]["model"],
        "client": QdrantClient(":memory:"),
        "embedding_model": "BAAI/bge-small-en-v1.5",
    },
)
```
code from: https://github.com/microsoft/autogen/blob/main/notebook/agentchat_qdrant_RetrieveChat.ipynb

**Comment by rickyloynd-microsoft** [comment]
@thinkall fyi

**Comment by thinkall** [comment]
Yes, but you need to extend the `RetrieveUserProxyAgent` like `QdrantRetrieveUserProxyAgent` did. Also, #619 will refactor the RAG agents, you may need to follow that once it will be merged.
